test/fetchHttpClientTests.ts

```diff
@@ -0,0 +1,83 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+import * as assert from "assert";
+import { fetchHttpClient } from "../lib/fetchHttpClient";
+import { HttpRequest } from "../lib/httpRequest";
+import { HttpResponse } from "../lib/httpResponse";
+
+describe("fetchHttpClient", () => {
+    it("should send HTTP requests", () => {
+        const request = new HttpRequest("GET", "http://www.example.com", {});
+        return fetchHttpClient(request)
+            .then((response: HttpResponse) => {
+                assert.deepStrictEqual(response.request, request);
```

Comment:
Do you have a preference toward asserts over `(2+2).should.equal(4)` or `expect(2+2).to.equal(4)` syntax?

Reply:
I definitely prefer assertions. I know it's commonplace in Javascript, but I'm not a fan on libraries injecting stuff into my types/objects.